Add segmented control component (#494)
microsoft/atlas-design: site/src/components/segmented-controls.md

---
title: Segmented Controls
description: Segmented controls component in the Atlas Design System
template: standard
classType: Component
classPrefixes:
  - segmented-control
  - segmented-controls

>   - segmented-controls-footer

Reviewer: nit: remove this now that `segmented-controls-footer` styles are out for now?

Reviewer: _(I'm comfortable with this change happening in a followup docs PR, if need)_

---

## Segmented Control

The segmented controls component is a tabs-like widget, containing a list of buttons for toggling an active view from a list. The view can be set by clicking the appropriate segmented control button, pressing the left or right arrow keys, or by clicking an optional previous or next button.

```html
<tab-container>
	<div class="segmented-controls" slot="tablist-wrapper">
		<button
			type="button"
			class="segmented-control-previous"
			title="Previous segment"
			data-tab-container-nav="previous"
		>
			<span class="icon" aria-hidden="true">
				<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 448">
					<path
						class="fill-current-color"
						d="M448 224H62.625l148.688 148.688-22.625 22.625L1.375 208 188.688 20.688l22.625 22.625L62.625 192H448v32z"
					/>
				</svg>
			</span>
		</button>

		<div class="segmented-control-list" role="tablist" aria-orientation="horizontal">
			<button
				class="segmented-control"
				type="button"
				id="sc-one-1"
				role="tab"
				aria-selected="true"

> 				data-segmented-control

Reviewer: _(I'm comfortable with this change happening in a followup docs PR, if need)_

> 				data-segmented-control

Reviewer 2: removing them now.

			data-segmented-control-nav="next"
		>
			<span class="icon" aria-hidden="true">
				<svg xmlns="http://www.w3.org/2000/svg" viewBox="-255 57 448 448">
					<path
						class="fill-current-color"
						d="M-253.625 281v-32H131.75L-16.938 100.313 5.687 77.688 193 265 5.687 452.313l-22.625-22.625L131.75 281h-385.375z"
					/>
				</svg>
			</span>
		</button>
	</div>
	<div class="margin-block-md text-align-center">
		<div role="tabpanel" aria-labelledby="sc-one-1" data-segmented-control-item="1">
			<p>Tab Item 1</p>
		</div>
		<div role="tabpanel" aria-labelledby="sc-one-2" data-segmented-control-item="2" hidden>
			<p>Tab Item 2</p>
		</div>
		<div role="tabpanel" aria-labelledby="sc-one-3" data-segmented-control-item="3" hidden>
			<p>Tab Item 3</p>
		</div>
		<div role="tabpanel" aria-labelledby="sc-one-4" data-segmented-control-item="4" hidden>
			<p>Tab Item 4</p>
		</div>
		<div role="tabpanel" aria-labelledby="sc-one-5" data-segmented-control-item="5" hidden>
			<p>Tab Item 5</p>
		</div>
		<div role="tabpanel" aria-labelledby="sc-one-6" data-segmented-control-item="6" hidden>
			<p>Tab Item 6</p>
		</div>
	</div>
</tab-container>
```

## Size modification

> Using the `segmented-controls-md` or `segmented-controls-lg` will increase the padding of the buttons.

Reviewer 3: Remove `segmented-controls-md`, we don't have that option available

---
title: Segmented Controls
description: Segmented controls component in the Atlas Design System
template: standard
classType: Component
classPrefixes:
  - segmented-controls
---

# Segmented Control

> Segmented control component consists of navigating to the previous or next segmented control item by using the arrow keys or clicking on a segmented control. The result will change the content to display the current selected item. To do this the `[data-segmented-control="1"]` and `[data-segmented-control-item="1"]` must have matching numbers as these act as the index for navigation.

Reviewer: > "To do this the `[data-segmented-control="1"]` and `[data-segmented-control-item="1"]` must have matching numbers as these act as the index for navigation."

This bit's no longer accurate, so these docs should be updated. Currently, a `[data-segmented-control]` attribute is required, since it's the selector we're using to get the list of tabs, but only in a boolean present/absent sense - it doesn't need any value at all, let alone any specific index.

Reviewer 2: This should be gone now.

```html
<tab-container data-segmented-controls-container>
	<div class="segmented-controls" slot="tablist-wrapper">
		<button
			type="button"
			class="segmented-control-previous"
			title="Previous segment"
			data-segmented-control-nav="previous"
		>
			<span class="icon" aria-hidden="true">
				<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 448">
					<path
						class="fill-current-color"
						d="M448 224H62.625l148.688 148.688-22.625 22.625L1.375 208 188.688 20.688l22.625 22.625L62.625 192H448v32z"
					/>
				</svg>
			</span>
		</button>
		<div class="segmented-control-list" role="tablist" aria-orientation="horizontal">
			<button
				class="segmented-control"
				type="button"
				id="sc-one-1"
				role="tab"
				aria-selected="true"

> 				data-segmented-control="1"

Reviewer: In keeping with the docs update comment above, we should update our example markup so as to not accidentally suggest any values are necessary/useful here:
```suggestion
				data-segmented-control
```
```suggestion
				data-segmented-control="1"
```

Reviewer 2: Should be fixed

				type="button"
				id="sc-one-2"
				role="tab"
				data-segmented-control="2"
			>
				Plan
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-one-3"
				role="tab"
				data-segmented-control="3"
			>
				Prepare
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-one-4"
				role="tab"
				data-segmented-control="4"
			>
				Adopt
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-one-5"
				role="tab"
				data-segmented-control="5"
			>
				Govern
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-one-6"
				role="tab"
				data-segmented-control="6"
			>
				Manage
			</button>
		</div>
		<button
			type="button"
			class="segmented-control-next"
			title="Next segment"
			data-segmented-control-nav="next"
		>
			<span class="icon" aria-hidden="true">
				<svg xmlns="http://www.w3.org/2000/svg" viewBox="-255 57 448 448">
					<path
						class="fill-current-color"
						d="M-253.625 281v-32H131.75L-16.938 100.313 5.687 77.688 193 265 5.687 452.313l-22.625-22.625L131.75 281h-385.375z"
					/>
				</svg>
			</span>
		</button>
	</div>

> 	<div role="tabpanel" aria-labelledby="sc-one-1" data-segmented-control-item="1">

Reviewer: Since `data-segmented-control-item` isn't being used at all anymore, let's make sure we remove these from our examples too:
```suggestion
	<div role="tabpanel" aria-labelledby="sc-one-1">
```
```suggestion
	<div role="tabpanel" aria-labelledby="sc-one-1" data-segmented-control-item="1">
```

Reviewer 2: Should be fixed.

		<div class="margin-block-md text-align-center">
			<p>Tab Item 1</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-2" data-segmented-control-item="2" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 2</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-3" data-segmented-control-item="3" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 3</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-4" data-segmented-control-item="4" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 4</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-5" data-segmented-control-item="5" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 5</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-6" data-segmented-control-item="6" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 6</p>
		</div>
	</div>
</tab-container>
```

## Size modification

Using the `segmented-controls-lg` will increase the padding of the buttons.

> ```html

Reviewer: This demo should be updated with the above feedback as well

---
title: Segmented Controls
description: Segmented controls component in the Atlas Design System
template: standard
classType: Component
classPrefixes:
  - segmented-control
  - segmented-controls
  - segmented-controls-footer
---

> ## Segmented Control

Reviewer 4: If you like, we can revisit this in a follow up PR, but I think this component documentation should probably be for the styling. The documentation here seems to be primarily for the JavaScript/Web Component aspect of it, which should be separate from the styling (since they are de-coupled). I think the doc for the JS should be in a separate folder, maybe called JS or web-components.

Reviewer 2: I will make sure this is my first follow up.

					<path
						class="fill-current-color"
						d="M-253.625 281v-32H131.75L-16.938 100.313 5.687 77.688 193 265 5.687 452.313l-22.625-22.625L131.75 281h-385.375z"
					/>
				</svg>
			</span>
		</button>
	</div>

	<div role="tabpanel" aria-labelledby="sc-one-1">
		<div class="margin-block-md text-align-center">
			<p>Tab Item 1</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-2" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 2</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-3" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 3</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-4" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 4</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-5" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 5</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-6" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 6</p>
		</div>
	</div>
</tab-container>
```

## Segmented Control with footer Next/Prev

> Removing the next and previous buttons will still provide a seamless experience on the tabs control. Small restructuring allows for the tabs-container component to allocate a footer to show after the tabpanel. The key difference is that of the `div.segmented-controls` taking on the `role=tablist` and `aria-orientation` attributes. This restructuring provides the grid responsiveness without buttons next to the control-list.

Reviewer 4: Docs are mixing styling with web component again.

Reviewer 2: Going to remove the footer nav altogether.   I will be immediately revisiting this and working towards a cleaner presentation and approach.  Might include some adjustments to the tab-container source code because its not accounting for things properly on the after panels.  So keep an eye out for that PR follow up.

				<svg xmlns="http://www.w3.org/2000/svg" viewBox="-255 57 448 448">
					<path
						class="fill-current-color"
						d="M-253.625 281v-32H131.75L-16.938 100.313 5.687 77.688 193 265 5.687 452.313l-22.625-22.625L131.75 281h-385.375z"
					/>
				</svg>
			</span>
		</button>
	</div>

	<div role="tabpanel" aria-labelledby="sc-one-1">
		<div class="margin-block-md text-align-center">
			<p>Tab Item 1</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-2" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 2</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-3" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 3</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-4" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 4</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-5" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 5</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-6" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 6</p>
		</div>
	</div>
</tab-container>
```

> ## Segmented Control with footer Next/Prev

Reviewer 4: I don't quite grok this part, do we need it?

Reviewer 2: do you mean the variant aspect of it?

Reviewer 4: And what are they?

Reviewer 2: Omitting for now, but will follow up with PR soon.

		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-6" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 6</p>
		</div>
	</div>

	<!-- After Tabpanels -->
	<div class="segmented-controls-footer border-top buttons padding-top-sm">
		<button class="button button-sm button-primary" data-segmented-control-nav="previous">
			<span class="icon" aria-hidden="true">
				<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 448">
					<path
						class="fill-current-color"
						d="M448 224H62.625l148.688 148.688-22.625 22.625L1.375 208 188.688 20.688l22.625 22.625L62.625 192H448v32z"
					/>
				</svg>
			</span>
			<span>Previous</span>
		</button>

		<button class="button button-sm button-primary button-filled" data-segmented-control-nav="next">
			<span>Next</span>
			<span class="icon" aria-hidden="true">
				<svg xmlns="http://www.w3.org/2000/svg" viewBox="-255 57 448 448">
					<path
						class="fill-current-color"
						d="M-253.625 281v-32H131.75L-16.938 100.313 5.687 77.688 193 265 5.687 452.313l-22.625-22.625L131.75 281h-385.375z"
					/>
				</svg>
			</span>
		</button>
	</div>
</tab-container>
```

## Size modification

> Using the `segmented-controls-lg` can be used to make segmented control bigger than the default.

Reviewer 4: This size seems like an issue on mobile. It's so big that the center button is usually cut off. Do we need it? Could we instead just ensure it scales the way we want from narrow to widescreens? (The default version seems to do this quite well).

Reviewer 2: I agree.

I think forcing it down in size to the standard, at mobile, would be a good compromise.  That way the larger views can still use the bigger version.

Reviewer 2: I went ahead and removed, based on the comps I dont see the reason for the large variant.  Rather just the nice responsive nature we already have.

						d="M448 224H62.625l148.688 148.688-22.625 22.625L1.375 208 188.688 20.688l22.625 22.625L62.625 192H448v32z"
					/>
				</svg>
			</span>
		</button>
		<div class="segmented-control-list" role="tablist" aria-orientation="horizontal">
			<button
				class="segmented-control"
				type="button"
				id="sc-two-1"
				role="tab"
				aria-selected="true"
				data-segmented-control
			>
				Define Strategy
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-two-2"
				role="tab"
				data-segmented-control
			>
				Plan
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-two-3"
				role="tab"
				data-segmented-control
			>
				Prepare
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-two-4"
				role="tab"
				data-segmented-control
			>
				Adopt
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-two-5"
				role="tab"
				data-segmented-control
			>
				Govern
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-two-6"
				role="tab"
				data-segmented-control
			>
				Manage
			</button>
		</div>
		<button
			type="button"

> 			class="segmented-control-next margin-left-xxs"

Reviewer 4: Don't think our examples/component should have margin atomics. let's handle this within the component itself rather than with atomics - same for the previous button: `segmented-control-previous margin-right-xxs`. Would be good to tighten up these gaps on mobile anyway:

<img width="166" alt="image" src="https://github.com/microsoft/atlas-design/assets/30843002/aa9ae670-342c-41d1-b99b-267b6428e772">

Reviewer 2: Looks like I missed some removals.  Ill check it out

Reviewer 2: This should be adjusted now

Reviewer 2: Removed the margin, but acknowledged that the space needs some magic in future.  Magic to include some css logic to reduce next/prev?  Will seek feedback on options here.

---
title: Segmented Controls
description: Segmented controls component in the Atlas Design System
template: standard
classType: Component

> classPrefixes:
>   - segmented-controls

Reviewer: Some classnames start with `segmented-control` now, so let's add that in:
```suggestion
classPrefixes:
  - segmented-control
  - segmented-controls
```

---

## Segmented Control

> Segmented control component consists of navigating to the previous or next segmented control item by using the arrow keys or clicking on a segmented control. The result will change the content to display the current selected item.

Reviewer: What do you think about this?

```suggestion
The segmented controls component is a tabslike widget, containing a list of buttons for toggling an active view from a list. The view can be set by clicking the appropriate segmented control button, pressing the left or right arrow keys, or by clicking an optional previous or next button.
```

					<path
						class="fill-current-color"
						d="M-253.625 281v-32H131.75L-16.938 100.313 5.687 77.688 193 265 5.687 452.313l-22.625-22.625L131.75 281h-385.375z"
					/>
				</svg>
			</span>
		</button>
	</div>

	<div role="tabpanel" aria-labelledby="sc-one-1">
		<div class="margin-block-md text-align-center">
			<p>Tab Item 1</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-2" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 2</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-3" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 3</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-4" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 4</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-5" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 5</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-one-6" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 6</p>
		</div>
	</div>
</tab-container>
```

## Segmented Control with footer Next/Prev

> Removing the next and previous buttons will still provide a seamless experience on the tabs control. Small restructuring allows for the tabs-container component to allocate the `after-tabpanel` footer.

Reviewer: Nit: the use of Markdown code syntax for `after-tabpanel` made me think `after-tabpanel` was some class or keyword I'd need to use to achieve the effect. Suggest rewording as:

```suggestion
Removing the next and previous buttons will still provide a seamless experience on the tabs control. Small restructuring allows for the tabs-container component to allocate a footer to show after the tabpanel.
```

			<p>Tab Item 2</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-3" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 3</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-4" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 4</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-5" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 5</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-6" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 6</p>
		</div>
	</div>

	<!-- After Tabpanels -->
	<div class="border-top display-flex padding-top-sm">
		<button
			type="button"
			class="button button-sm button-primary margin-right-sm"
			title="Previous segment"
			data-segmented-control-nav="previous"
		>
			<span class="icon" aria-hidden="true">
				<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 448">
					<path
						class="fill-current-color"
						d="M448 224H62.625l148.688 148.688-22.625 22.625L1.375 208 188.688 20.688l22.625 22.625L62.625 192H448v32z"
					/>
				</svg>
			</span>
			Previous
		</button>

> 		<button

Reviewer: I can't make out a focus outline on this "Next" button. I also can't find a clear reason why that is, even on a cursory look through DevTools.

Reviewer: On further investigation, the outline *does* appear, it's just showing as black on the dark mode's dark background. 

I think this is a case of our `outline-color: inherit` failing to break through the shadow DOM barrier.

Reviewer 2: Right and because the buttons are outside of that `segmented-controls` class it wont catch the outline.  Might need an atomic here.

Reviewer: LGTM

---
title: Segmented Controls
description: Segmented controls component in the Atlas Design System
template: standard
classType: Component
classPrefixes:
  - segmented-control
  - segmented-controls

>   - segmented-controls-after

Reviewer: Suggest either updating `segmented-controls-after` to `segmented-controls-footer`, or just removing altogether.

Reviewer 2: ✅

---

## Segmented Control

The segmented controls component is a tabs-like widget, containing a list of buttons for toggling an active view from a list. The view can be set by clicking the appropriate segmented control button, pressing the left or right arrow keys, or by clicking an optional previous or next button.

```html

> <tab-container data-segmented-controls-container>

Reviewer 3: I feel like we need some explanation around `tab-container` usage somewhere here. We haven't re-used any custom components before, so I see scenarios when people might get confused about these examples and not really understanding why it's here. @BenDMyers @homing1 @wibjorn any thoughts on this?

Reviewer: I agree that adding further explanation would be good here. Such explanation should make it clear that `tab-container` is responsible for handling keyboard support and ARIA attributes out of the box.

I think this explanation should also link to `tab-container`'s documentation.

Reviewer 2: Could this be non-blocking to this release?  Or would you prefer that I compile something asap?

Reviewer 3: I really understand your eagerness to get this shipped (I've been in these shoes many times), but we have to make this right from the first round. As soon as it merged it becomes available to anyone and re-visiting later might cause issues in case component is already in use.

Also, the more I think about it - more I'm concerned about adding tab-container to Atlas. Atlas is a styling framework and introducing dependency on someone else's custom component we have no control over and adding extra JS to make it fit _our_ need bothers me. Should we move all the functionality logic to docs-ui as we normally do and have only styling in Atlas? Think about breadcrumbs component. it's very similar thing

@wibjorn @BenDMyers @homing1 please share your thoughts on this.

Reviewer 4: I'm not necessarily against using GH's tab-container as a base for this. Atlas does have a minimal-ist approach to JS, which is to say we only add it when we need to ensure an interaction is usable and accessible - but prefer the CSS only approach first.

I'm more interested in the _how_ here, since that will inform us of the best approach.

Reviewer: I think `tab-container` makes a good base for the logic, and I'd be generally in favor of continuing to use it. Accessible tabs will require JS -- CSS-only isn't sufficient -- and `tab-container` gives us most of what we need in a well-tested way.

My two gripes with `tab-container`, from having been involved in debugging the styles and refactoring the scripts are:

1. Many of its methods, which would have been useful for its previous/next button augmentation are instead private, which means some reverse engineering.
1. Its opinionated use of slots means markup inside generally has to be pretty specific, which is what's led to some of the specific nesting you see here. (Also, the use of shadow DOM has posed some annoying problems for our `outline-color` inheritance)

Reviewer 4: My preference is to keep the JS here, since it's required for a usable experience. If we do run into major issues with private method, etc, then we can copy the code in an begin making our changes.

				type="button"
				id="sc-three-5"
				role="tab"
				data-segmented-control
			>
				Govern
			</button>
			<button
				class="segmented-control"
				type="button"
				id="sc-three-6"
				role="tab"
				data-segmented-control
			>
				Manage
			</button>
		</div>
	</div>

	<div role="tabpanel" aria-labelledby="sc-three-1">
		<div class="margin-block-md text-align-center">
			<p>Tab Item 1</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-2" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 2</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-3" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 3</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-4" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 4</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-5" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 5</p>
		</div>
	</div>
	<div role="tabpanel" aria-labelledby="sc-three-6" hidden>
		<div class="margin-block-md text-align-center">
			<p>Tab Item 6</p>
		</div>
	</div>

	<!-- After Tabpanels -->

> 	<div class="segmented-controls-footer border-top display-flex padding-top-sm">

Reviewer 3: `display-flex` -> `buttons`

		<button

> 			class="button button-sm button-primary margin-right-sm"

Reviewer 3: Remove margin atomic, `.buttons` on parent element will handle the spacing

> 			title="Previous segment"

Reviewer 3: Is `title` needed here? Same question for the next button. These two aren't icon only button so probably should be removed.

Reviewer: **I'd be in favor of this.** A speech control user would, based on the visual text, expect to say "click Previous" to click the "Previous"/"Previous segment" button, but `title` could make it so that some speech control software is specifically expecting "click Previous segment," with no clear visual indicator of that expectation. If we're showing some visual text, it's better to not also have hidden text that could sway the accessible name, to make targeting this button more predictable.

			data-segmented-control-nav="previous"
		>
			<span class="icon" aria-hidden="true">
				<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 448">
					<path
						class="fill-current-color"
						d="M448 224H62.625l148.688 148.688-22.625 22.625L1.375 208 188.688 20.688l22.625 22.625L62.625 192H448v32z"
					/>
				</svg>
			</span>

> 			Previous

Reviewer 3: Wrap text into `span` to fix the spacing issue between icon and text. Same on "Next" button

![image](https://github.com/microsoft/atlas-design/assets/57374379/3d20d706-4f57-4458-9876-cfe6313ece45)

		</button>

		<button

> 			class="button button-primary button-filled"

Reviewer 3: Missing `button-sm` to match the "Previous" button's size

			title="Next segment"
			data-segmented-control-nav="next"
		>
			Next
			<span class="icon" aria-hidden="true">
				<svg xmlns="http://www.w3.org/2000/svg" viewBox="-255 57 448 448">
					<path
						class="fill-current-color"
						d="M-253.625 281v-32H131.75L-16.938 100.313 5.687 77.688 193 265 5.687 452.313l-22.625-22.625L131.75 281h-385.375z"
					/>
				</svg>
			</span>
		</button>
	</div>
</tab-container>
```

## Size modification

> Using the `segmented-controls-lg` will increase the padding of the buttons.

Reviewer 3: it's not only about the padding, the font size increases too. What about this:
> `segmented-controls-lg` can be used to make segmented control bigger than the default.
